docs/direnv.md

```diff
@@ -0,0 +1,154 @@
+
+```bash
+# Bash
+echo 'eval "$(direnv hook bash)"' >> ~/.bashrc
+
+# Zsh
+echo 'eval "$(direnv hook zsh)"' >> ~/.zshrc
+
+# Fish
+echo 'direnv hook fish | source' >> ~/.config/fish/config.fish
+```
+
+Remember to restart your shell or open a new terminal window to apply these changes.
+
+Next, you'll need to enable Direnv for your project directories. When you navigate into a directory with a `.envrc` file, Direnv will block the environment until you approve the changes by running `direnv allow`.
+
+### Pitfalls
+
+Direnv can be a bit tricky to get working. Some common issues include:
+
+- Forgetting to add the Direnv hook to your shell's configuration file.
+- Navigating to a directory before you've approved its `.envrc` file.
+- Forgetting to restart your shell or open a new terminal after setting up Direnv.
+
+Be sure to follow the installation and setup instructions carefully, and consult the [Direnv documentation](https://direnv.net/docs/troubleshooting.html) if you encounter problems.
+
+## Set Up Your Haskell Project
+
+Let's start by creating a simple HelloWorld Haskell application. First, navigate to your preferred working directory, then:
+
+```bash
+$ mkdir HelloWorld
+$ cd HelloWorld
+$ cabal init --is-executable
+```
+
+This will generate a basic `Main.hs` file with a HelloWorld program.
+
+## Configure Direnv for Your Project
+
+Create a `.envrc` file in your project root with the following content:
+
+```bash
+use flake "github:input-output-hk/devx#ghc8107-iog"
+```
+
+Then, allow Direnv to take effect:
+
+```bash
+$ direnv allow
+```
+
+Direnv will use the specified flake to provide a shell with all the Haskell development tools you need, as configured in the Devx repository.
+
+## Setting Up Your Editor
+
+### Visual Studio Code (VSCode)
+
+First, ensure that you have the following extensions installed:
+
+- [Nix Extension Pack](https://marketplace.visualstudio.com/items?itemName=pinage404.nix-extension-pack)
+- [Haskell](https://marketplace.visualstudio.com/items?itemName=haskell.haskell)
+
+The Nix Extension Pack brings Direnv support, while the Haskell extension provides support for the Haskell Language Server.
+
+Upon opening the folder `myproject` in VSCode, Direnv should automatically load your environment, and Haskell Language Server should be able to provide features like type checking, linting, and autocompletion.
+
+
+When using Direnv with VSCode, it's crucial to have the shell pre-cached correctly. VSCode can end up loading plugins before Direnv has had a chance to modify the PATH if the shell is not pre-cached. 
+
+This can be solved by launching VSCode from the terminal within the Direnv environment:
+
+```bash
+$ code .
+```
+
+This way, VSCode inherits the correct environment from the shell.
```

Comment:
Does this always work across _all_ platforms? I'm not sure `code` is even trivially available on macOS. Nor on linux where VSCode is installed outside of the package manager?

I _do_ think restarting vscode or reloading the window will fix this once the shell has been instantiated.